.github/workflows/docker.yml

```diff
@@ -0,0 +1,15 @@
+name: Docker
+on: push
+jobs:
+  build:
+    name: Publish to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Publish
+        uses: whoan/docker-build-with-cache-action@v3.1.1
```

Comment:
Why not docker's [own action](https://github.com/docker/build-push-action)?

Reply:
Never knew this existed. I will make changes then.

Reply:
Done.